api/src/test/java/io/grpc/ForwardingExecutorService.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Forwards all methods to delegate.
+ */
+abstract class ForwardingExecutorService implements ExecutorService {
```

Comment:
Since this is in the `test` source set, does it make sense to just go ahead and use Guava? I guess dependencies aren't as much of an issue in the `test` scope.

Reply:
Doh, yes. I checked Guava for ForwardingScheduledExecutorService and so started making my own, forgetting that Guava has ForwardingExecutorService.